src/client/action/list_databases.rs

```diff
@@ -0,0 +1,158 @@
+use std::{future::IntoFuture, marker::PhantomData};
+
+use bson::{Bson, Document};
+use futures_util::FutureExt;
+
+#[cfg(any(feature = "sync", feature = "tokio-sync"))]
+use crate::sync::Client as SyncClient;
+use crate::{
+    client::BoxFuture,
+    error::{ErrorKind, Result},
+    operation::list_databases as op,
+    results::DatabaseSpecification,
+    Client,
+    ClientSession,
+};
+
+use super::option_setters;
+
+impl Client {
+    /// Gets information about each database present in the cluster the Client is connected to.
+    pub fn list_databases(&self) -> ListDatabases {
+        ListDatabases {
+            client: self,
+            options: Default::default(),
+            session: None,
+            mode: PhantomData,
+        }
+    }
+
+    /// Gets the names of the databases present in the cluster the Client is connected to.
+    pub fn list_database_names(&self) -> ListDatabases<'_, Names> {
+        ListDatabases {
+            client: self,
+            options: Default::default(),
+            session: None,
+            mode: PhantomData,
+        }
+    }
+}
+
+#[cfg(any(feature = "sync", feature = "tokio-sync"))]
+impl SyncClient {
+    /// Gets information about each database present in the cluster the Client is connected to.
+    pub fn list_databases(&self) -> ListDatabases {
+        self.async_client.list_databases()
+    }
+
+    /// Gets the names of the databases present in the cluster the Client is connected to.
+    pub fn list_database_names(&self) -> ListDatabases<'_, Names> {
+        self.async_client.list_database_names()
+    }
+}
+
+/// Gets information about each database present in the cluster the Client is connected to.
+#[must_use]
+pub struct ListDatabases<'a, M: Mode = Full> {
```

Comment:
Good call - you're right, it's not actually needed.  Removed it and updated the names of the unit structs to make their purpose more clear.  (That also led me on a tangent to update the rustdoc so that the usage pattern should be clear.)